### PR TITLE
Do not run jobs when manual_trigger=True

### PR DESCRIPTION
**Pull request by Frawless**
This PR adds logic that when user specify `manual_trigger: True` in job definition, the handler will not consider this job as executable automatically, only via comment.

TODO:

- [x] Write new tests or update the old ones to cover new functionality.
- [ ] Update or write new documentation in `packit/packit.dev`.

Fixes https://github.com/packit/packit-service/issues/1806

Related to https://github.com/packit/packit/pull/1984

Merge after https://github.com/packit/packit/pull/1984

---

RELEASE NOTES BEGIN
￼ TODO
RELEASE NOTES END

### CODE REVIEW

File: packit_service/worker/jobs.py
Comment (lbarcziova) [inline packit_service/worker/jobs.py:618]: I assume there needs to be some additional condition, something like:
```python
if not job.manual or self.event in MANUAL_EVENTS:
```

(where the MANUAL_EVENTS would be list of events for comments/check re-runs)

But once we have a testcase, that will be clearer

**Review by lbarcziova — COMMENTED** [review]
thanks again for the contribution!

File: packit_service/worker/jobs.py
Comment (lbarcziova) [inline packit_service/worker/jobs.py:634]: as I am looking into the list of MANUAL_EVENTS, we could reduce the events to only `AbstractCommentEvent`, and `CheckRerunEvent` there and use `any(isinstance(self.event, event_type) for event_type in MANUAL_EVENTS)` here to count with inheritance

File: packit_service/worker/handlers/testing_farm.py
Comment (lbarcziova) [inline packit_service/worker/handlers/testing_farm.py:81]: we want to react only to pull request comments for running the Testing Farm

**Review by lbarcziova — COMMENTED** [review]
regarding the tests, I would add few test cases [here](https://github.com/packit/packit-service/blob/38d6158d1b718a079d4b08c35bbd69caf2e7859b/tests/unit/test_jobs.py#L2899)... if you need help with that, we can do that :) 


we also run pre-commit in the CI that enforces rebased branch, [see](https://github.com/packit/contributing#rebasing)

File: packit_service/worker/handlers/testing_farm.py
Comment (Frawless) [reply packit_service/worker/handlers/testing_farm.py:81]: I understood that some of the users want to trigger it from the issues. Is that something we will deal with in another pr? Or I just misunderstood it?

Comment (lbarcziova) [reply packit_service/worker/handlers/testing_farm.py:81]: we allow retriggering from issues for the [`propose_downstream` job](https://packit.dev/docs/configuration/#propose_downstream)

File: packit_service/worker/jobs.py
Comment (Frawless) [reply packit_service/worker/jobs.py]: Ye just leftover from debugging, removed!

File: packit_service/worker/jobs.py
Comment (Frawless) [reply packit_service/worker/jobs.py]: Done :)

Comment (lbarcziova) [reply packit_service/worker/jobs.py]: is the `job.trigger != JobConfigTriggerType.pull_request` acutally needed?

Comment (Frawless) [reply packit_service/worker/jobs.py]: in case user will specify `manual_trigger=True` in jobs that do not care about the option we should skip it...now I am thinking if checking job type is a better option or not 🤔  something like `job.type != JobType.tests`...or is this all handled by decorators here https://github.com/packit/packit-service/blob/38d6158d1b718a079d4b08c35bbd69caf2e7859b/packit_service/worker/handlers/testing_farm.py#L70 ?

Comment (lbarcziova) [reply packit_service/worker/jobs.py]: yes, the decorators should take care of the mapping itself

my thinking was that by omitting that part of the condition this could be used generally for any job (e.g. also for the mentioned propose-downstream) without binding it to the testing farm job (+ the docs would explain the option properly); @mfocko WDYT?

Comment (mfocko) [reply packit_service/worker/jobs.py]: +1 for having it for all commands; if it gets used in a job that is not easily run via comment, it's a user error IMO, not an issue :shrug:

Also I looked around, not only at the diff, I would probably merge it with the `if` in which it's wrapped now; keeping the last check of the `if` above at the end even after merging it in one condition. (It just seems like an unnecessary indentation)

Comment (Frawless) [reply packit_service/worker/jobs.py]: Got it, I updated the code and tests.

**Review by lbarcziova — COMMENTED** [review]
LGTM once the tests are green :rocket:

**Comment by lbarcziova** [comment]
recheck

**Review by lbarcziova — APPROVED** [review]
nice, thanks!!

could you please squash the commits here as well?

**Comment by Frawless** [comment]
@lbarcziova done, thanks

**Pull request merged**
